x-pack/plugins/uptime/public/components/synthetics/check_steps/expand_row_col.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiButtonIcon } from '@elastic/eui';
+import { Ping } from '../../../../common/runtime_types/ping';
+import { ExecutedStep } from '../executed_step';
+
+interface Props {
+  ping: Ping;
+  browserConsole?: string;
+  expandedRows: Record<string, JSX.Element>;
+  setExpandedRows: (val: Record<string, JSX.Element>) => void;
+}
+
+export const toggleExpand = ({
+  ping,
+  browserConsole = '',
+  expandedRows,
+  setExpandedRows,
+}: Props) => {
+  // If already expanded, collapse
+  if (expandedRows[ping.docId]) {
```

Comment:
So you can have multiple expanded rows at one time?

Reply:
Yup, this is how ping list work as well.

Reply:
We used to only have 1, then we allowed `n`, so this is consistent with what we have today. It's always a conversation we could revisit.